src/cmd-buildextend-aliyun

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env python3
+# NOTE: PYTHONUNBUFFERED is set in cmdlib.sh for unbuffered output
+#
+# An operation that mutates a build by generating an aliyun image.
+
+import os
+import sys
+import json
+import shutil
+import argparse
+import subprocess
+
+sys.path.insert(0, os.path.dirname(os.path.abspath(__file__)))
+from cosalib.builds import Builds
+from cosalib.cmdlib import run_verbose, write_json, sha256sum_file, get_basearch
+
+# Parse args and dispatch
+parser = argparse.ArgumentParser()
+parser.add_argument("--build", help="Build ID")
+parser.add_argument("--region", help="Cloud Region")
+parser.add_argument("--bucket", help="OSS Bucket")
+parser.add_argument("--log-level", help="ore log level")
+parser.add_argument("--upload", action='store_true', default=False,
+                    help="Upload to Aliyun")
+args = parser.parse_args()
+
+# Identify the builds and target the latest build if none provided
+builds = Builds()
+if not args.build:
+    args.build = builds.get_latest()
+print(f"Targeting build: {args.build}")
+
+if args.upload:
+    for attr in ['region', 'bucket']:
+        if getattr(args, attr) is None:
+            raise Exception(f"Must provide --{attr} option in order to to upload")
```

Comment:
Minor: `to to` (I'm guessing that's just inherited from the AWS code :) )

Reply:
Fixed 😄